15 LoginForm/src/pages/login/form.tsx

```diff
@@ -1,37 +1,35 @@
 import * as React from "react"
 import {hashHistory} from 'react-router'
 import {LoginEntity} from '../../model/login';
+import {FormField} from '../common/formField';
 
 interface Props {
    loginInfo : LoginEntity;
    updateLoginInfo : (loginInfo : LoginEntity) => void;
    performLogin : () => void;
 }
 
-export const Form = (props: Props) => {
+export class Form extends React.Component<Props, {}> {
```

Comment:
I added the private function:
'''javascript
updateFieldValue(fieldName: string, fieldValue: any){
    const newLoginEntity = this.props.loginInfo;
    newLoginEntity[fieldName] = fieldValue;
    this.props.updateLoginInfo(newLoginEntity);
  }
''''
Can that be done without promoting it to class?